src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/FixedWindowRateLimiter.cs

```diff
@@ -44,7 +44,28 @@ public sealed class FixedWindowRateLimiter : ReplenishingRateLimiter
         /// <param name="options">Options to specify the behavior of the <see cref="FixedWindowRateLimiter"/>.</param>
         public FixedWindowRateLimiter(FixedWindowRateLimiterOptions options)
         {
-            _options = options ?? throw new ArgumentNullException(nameof(options));
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+            if (options.PermitLimit <= 0)
+            {
+                throw new ArgumentException($"{nameof(options.PermitLimit)} must be set to a value greater than 0.", nameof(options));
+            }
+            if (options.QueueLimit < 0)
+            {
+                throw new ArgumentException($"{nameof(options.QueueLimit)} must be set to a value greater than or equal to 0.", nameof(options));
+            }
+
+            _options = new FixedWindowRateLimiterOptions
+            {
+                PermitLimit = options.PermitLimit,
+                QueueProcessingOrder = options.QueueProcessingOrder,
+                QueueLimit = options.QueueLimit,
+                Window = options.Window,
```

Comment:
`Window` isn't being validated

Reply:
I removed that, there are a bunch of test cases where `Window` is `TimeSpan.Zero` that don't expect it to throw, so I figured it was a valid scenario

Reply:
> so I figured it was a valid scenario

It is (for now). But we should still validate that it isn't negative.

Reply:
Fixed + updated `InvalidOptionsThrows` tests